components/scream/src/share/io/scream_output_manager.cpp

```diff
@@ -111,30 +111,38 @@ setup (const ekat::Comm& io_comm, const ekat::ParameterList& params,
   // require it, we need to restart the output history.
   // E.g., we might save 30-day avg value for field F, but due to job size
   // break the run into three 10-day runs. We then need to save the state of
-  // our averaging in a "restart" file (e.g., the current avg and avg_count).
+  // our averaging in a "restart" file (e.g., the current avg).
   // Note: the user might decide *not* to restart the output, so give the option
   //       of disabling the restart. Also, the user might want to change the
   //       casename, so allow to specify a different casename for the restart file.
   if (m_is_restarted_run) {
     // Allow to skip history restart, or to specify a casename for the restart file
     // that is different from the casename of the current output.
     auto& restart_pl = m_params.sublist("Restart");
-    bool perform_history_restart = has_restart_data && restart_pl.get("Perform Restart",true);
+    bool perform_history_restart = restart_pl.get("Perform Restart",true);
```

Comment:
What if we don't have history restart data?  For example, a typical restart run in E3SM would be with monthly restarts and monthly average data.  Assuming that our run finishes normally, we wouldn't need to restart the history file since we would be starting a new month.  Doesn't this check make sure we don't throw an error in this case?

Reply:
This PR only fixes the issue when avg_type for history output is Instant. Fixes for multiple places will be needed for the other avg_type or output frequency units (such as monthly).

Reply:
Good point. We can check also the value of `m_output_control.nsteps_since_last_write` a few lines below before attempting to read data.

Reply:
@AaronDonahue The commit I just pushed should fix this.